### PR TITLE
refactor & corner support (2nd method).

### PR DESCRIPTION
**Pull request by fky2015**
If you need to modify the code based on this PR, you can merge it.

Closes FL-20.

### CODE REVIEW

File: src/adt/image.rs [hunk 1]
```diff
@@ -36,33 +34,60 @@ impl PositionOptionT for Image {
...
+
+        clip_path.append_child(rect);
+
+        defs.append_child(clip_path);
+        svg.append_child(defs);
+        let mut img = Element::new("image");
+        img.set_attr("href", self.path.clone());
+        img.set_attr("width", "100%");
+        img.set_attr("height", "100%");
+        img.set_attr("clip-path", "url(#clip)");
```
Comment (phy25) [inline src/adt/image.rs:60]: If we have multiple images, the ID `#clip` would need to be unique for each one of them. I am going to create a follow-up ticket (FL-24).

File: src/adt/tests.rs [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 use elementtree::Element;
 
-use super::foundation::Color;
-use super::svg::Canvas;
 use super::background::Background;
+use super::foundation::Color;
 use super::image::Image;
+use super::svg::Canvas;
 
 pub fn compare_svg_text(left: &str, right: &str) -> Result<()> {
```
Comment (phy25) [inline src/adt/tests.rs:9]: Looks like we are never going to use `compare_svg_text`. I am going to remove it at some point.

File: src/adt/image.rs [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
 use serde::{Deserialize, Serialize};
 
-use super::foundation::{Size, SizeOptionT, SizeOption, Position, PositionOptionT, PositionOption};
+use super::foundation::{Position, PositionOption, PositionOptionT, Size, SizeOption, SizeOptionT};
 use super::svg::SvgTangibleObject;
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Image {
     path: String,
+    round: Option<usize>,
```
Comment (phy25) [inline src/adt/image.rs:10]: `round` sounds like a common property to elements. When we implement this for a second element we need to figure out how to generalize the implementation better (in my unfortunate experience, people tend to forget this and just copy-paste code later, but I trust this project better :) ).

Comment (phy25) [reply src/adt/image.rs:10]: We also need to figure out why test coverage is complaining here, weird.

**Review by phy25 — APPROVED** [review]

File: src/adt/image.rs (hunk 1 above)
Comment (fky2015) [reply src/adt/image.rs:60]: > If we have multiple images, the ID `#clip` would need to be unique for each one of them. I am going to create a follow-up ticket (FL-24).

Sorry, but there is already a ticket [FL-23](https://linear.app/footlights/issue/FL-23/%E6%B8%B2%E6%9F%93%E6%97%B6%E7%BB%99%E6%AF%8F%E4%B8%AAyuan%E7%B4%A0%E5%A2%9E%E5%8A%A0%E7%BC%96%E5%8F%B7).

Comment (phy25) [reply src/adt/image.rs:60]: Closing mine as a duplicate. I think it's sort of fair to add these TODO as a code comment

**Pull request merged**